CSS.supports returns false for custom properties

rniwa opened the pull request:
#### 69362356a5abb6fcd0c1a11de27b42ecf85e6b7b
<pre>
CSS.supports returns false for custom properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=154669">https://bugs.webkit.org/show_bug.cgi?id=154669</a>

Reviewed by Alexey Proskuryakov.

Fix the bug that the version of DOMCSSNamespace::supports which takes property ID and value separately
couldn&apos;t parse CSS custom properties.

* LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/js/CSS-supports-L3-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/conditional-rules-expected.txt:
* Source/WebCore/css/DOMCSSNamespace.cpp:
(WebCore::DOMCSSNamespace::supports):

Canonical link: <a href="https://commits.webkit.org/252987@main">https://commits.webkit.org/252987@main</a>
</pre>

aproskuryakov commented on Source/WebCore/css/DOMCSSNamespace.cpp:
```diff
@@ -58,9 +58,16 @@ static String valueWithoutImportant(const String& value)
+    CSSPropertyID propertyID = cssPropertyID(propertyNameWithoutWhitespace);
+    if (propertyID == CSSPropertyInvalid && isCustomPropertyName(propertyNameWithoutWhitespace)) {
+        auto dummyStyle = MutableStyleProperties::create();
+        constexpr bool important = false;
```
I suggest calling this variable either `importance` or `notImportant`, to make the next line more readable. Currently, it says `important`, which is the exact opposite of what it is.

rniwa replied:
Good point. Will do.

rniwa committed:
CSS.supports returns false for custom properties
https://bugs.webkit.org/show_bug.cgi?id=154669

Reviewed by Alexey Proskuryakov.

Fix the bug that the version of DOMCSSNamespace::supports which takes property ID and value separately
couldn't parse CSS custom properties.

* LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/js/CSS-supports-L3-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/conditional-rules-expected.txt:
* Source/WebCore/css/DOMCSSNamespace.cpp:
(WebCore::DOMCSSNamespace::supports):

Canonical link: https://commits.webkit.org/252987@main
